### PR TITLE
Convert tests to pytest

### PR DESCRIPTION
**Pull request by rspeer**
This converts the testing framework of ConceptNet from (nosetests + awful shell scripts) to pytest.

- nose helpers such as `eq_` are replaced with asserts, of course
- Use fixtures instead of globals
- A fixture runs the test build for all tests that need it, unless `--quick` is passed
- A "mark" determines that tests that require the fully-built DB run only when `--fulldb` is passed
- The full-build tests were broken by the change to expand_terms, and many of them don't even require the full build anymore, so those are fixed now

### CODE REVIEW

File: conceptnet5/tests/small_build/test_lemmatizer.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from conceptnet5.language.lemmatize import lemmatize
+from conceptnet5.tests.conftest import run_build
+
+
+def test_lemmatize(run_build):
```
Comment (jlowryduda) [inline conceptnet5/tests/small_build/test_lemmatizer.py]: I think this could be a parametrized test.

File: conceptnet5/tests/test_vectors.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    assert expected_expanded_terms == expanded_terms
+
+
+def test_similar_terms(simple_frame):
+    """
+    Check if VectorSpaceWrapper's index is sorted and its elements are concepts.
+    """
+    vectors = VectorSpaceWrapper(frame=simple_frame)
+    vectors.load()
+    print(vectors.similar_terms('/c/en/figure skating', limit=3))
```
Comment (jlowryduda) [inline conceptnet5/tests/test_vectors.py]: A leftover comment?

File: conftest.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        '--quick', action='store_true', help='Use existing input files in tests/data'
+    )
+    parser.addoption(
+        '--fulldb', action='store_true', help='Test the fully-built conceptnet database'
+    )
+
+
+# https://docs.pytest.org/en/latest/example/simple.html#control-skipping-of-tests-according-to-command-line-option
+def pytest_configure(config):
+    config.addinivalue_line("markers", "requires_full_build: test requires the actual database to be built")
```
Comment (jlowryduda) [inline conftest.py]: Super nitpick: single quotes.

Comment (rspeer) [reply conftest.py]: The Daylight style guide isn't our style guide.

Comment (rspeer) [reply conftest.py]: I don't think we should spend any time caring about which kind of quotes we use, especially in copied code.

File: conftest.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--fulldb"):
+        # --runslow given in cli: do not skip slow tests
```
Comment (jlowryduda) [inline conftest.py]: Should `--runslow` in the comment have been `--fulldb`?

**Comment by rspeer** [comment]
Okay, I think I've addressed everything so far.

**Pull request merged**